modules/ingester/flush.go

```diff
@@ -150,12 +213,7 @@ func (i *Ingester) flushUserTraces(userID string) error {
 		return fmt.Errorf("instance id %s not found", userID)
 	}
 
-	for {
-		block := instance.GetBlockToBeFlushed()
-		if block == nil {
-			break
-		}
-
+	for _, block := range instance.GetBlocksToBeFlushed() {
```

Comment:
`GetBlocksToBeFlushed()` will only be called once.  If another block is added to the instances flush list during this time period it will not be picked up.  we need to repeatedly call this method on every loop in case more blocks get added.